### ISSUE TITLE
Godot 3.5 & 4.0 crash from time to time

### ISSUE THREAD

**Issue by denis-games** [godotengine/godot#78792]
### Godot version

v3.5.2.stable.official [170ba337a], v4.0.3.stable.official [5222a99f5]

### System information

Ubuntu 23.04, Windows 11

### Issue description

From time to time, several times per hour, Godot is shutting down without any reasons or warnings.
It could happened when saving a file or importing an assets (or anything else...).

### Steps to reproduce

I don't know how to reproduce it because it seems erratic.

I don't know, as weel, how to provide more information.

### Minimal reproduction project

N/A

**Comment by Calinou** [comment]
Closing, as feature proposals should be discussed on the [Godot proposals](https://github.com/godotengine/godot-proposals) repository. This repository is now used for bug reports only.

Make sure to follow the proposal template when opening a new proposal – don't just copy and paste the text you wrote in this issue.

**Comment by denis-games** [comment]
I'm testing it with godot-4.0.3-editor-debug-linux-gcc9.3 since yesterday, and I didn't have crash since then.

**Comment by denis-games** [comment]
I have the same issues with v4.1.rc2.official [46424488e].

And sometimes, the project won't run and when "reload current project", the editor quit without restarting.

Does the 4.1 will come with editor degub?

**Comment by AThousandShips** [comment]
> And sometimes, the project won't run and when "reload current project", the editor quit without restarting.

Already tracked in #74549

**Comment by denis-games** [comment]
This issues occurs several times per hour. I'm using Ubuntu 23.04 with Godot 4.1.1 stable ans Wayland.

I will try to use Xorg to see if the issue still happens.

@Calinou do you plan to release a debug release for 4.1 or 4.2 branch?

**Comment by Calinou** [comment]
> @Calinou do you plan to release a debug release for 4.1 or 4.2 branch?

Yes, but it may take a while to happen as I have to build it manually.

**Comment by denis-games** [comment]
After some more testing, it seems to occur only on old hardware. I'm currently using Godot v4.1.1.stable.official [bd6af8e0e].

It crashes a lot on my old Dell Inspiron 15 from 2013, but on the new one from 2022, I still have no issues.

**Comment by Calinou** [comment]
> It crashes a lot on my old Dell Inspiron 15 from 2013, but on the new one from 2022, I still have no issues.

What CPU and graphics card model does your 2013 laptop have? Are its drivers fully up-to-date from AMD/Intel/NVIDIA's website (not Windows Update, not Dell)?

**Comment by BiSerge** [comment]
I am using an ASUS K750JB laptop
Intel(R) Core(TM) i7-4700HQ CPU @ 2.40GHz   2.40 GHz
Video:
Intel(R) HD Graphics 4600
NVIDIA GeForce GT 740M
Windows 10 Home for one language

From time to time, several times an hour, Godot shuts down without any reason or warning. This happens at any time, when performing any operations. Sometimes this happens during prolonged downtime, without performing any actions.

I downloaded godot-3.5.1-editor-debug-windows-msvc2019 to find out the reasons. It also turns off without any reason or warning.

**Comment by BiSerge** [comment]
I used "Event Viewer" to identify the problem. Saw it:

A TDR has been detected.
The application must close.
Error code: 7
 (pid=1560 tid=1376 godot_v3.5.2-stable_win64.exe 64bit)

The video adapter driver has been updated. The solution proposed on the Internet, to add the TdrDelay parameter to the registry, did not help me. Please help me solve the problem.

**Comment by Calinou** [comment]
@BiSerge Which rendering method are you using in the project? (Only the Compatibility rendering method is supported on your hardware, but I'm asking anyway.)

Are you running Godot on the integrated or dedicated GPU? Try switching around the GPU you're using to run Godot.

**Comment by BiSerge** [comment]
@Calinou Thanks for the help.
I am using Godot v3.5.2. If I understood correctly, only OpenGL ES 2.0 and OpenGL ES 3.0 are in it. I thought this only affects the application being created, but not the editor.
I have configured a discrete NVIDIA GeForce GT 740 GPU for Godot.
I'll try your advice later.

**Comment by BiSerge** [comment]
Just think about how many people with old computers could face this problem and refuse to use Godot, not wanting to deal with the problem and write about the problem. I think you can help not only me, but also many, many others.
I want to clarify that I will not experience any problems with any other software on my computer. Which leads me to think that Godot is the problem. For example, Unity3D on my computer works without problems.

**Comment by BiSerge** [comment]
@Calinou Unfortunately, nothing has helped so far.
Godot always runs only NVIDIA GeForce.
OpenGL ES 2.0 or OpenGL ES 3.0 do not affect the stability of the work.
If you specify the TdrDelay parameter in the registry, Godot shuts down stably within 5 minutes.
If the TdrDelay parameter is deleted, Godot shuts down after different time intervals, it may work for 5 minutes, maybe 30.

**Comment by BiSerge** [comment]
@denis-games Could you connect to the solution of the problem, conduct tests and either confirm or refute my conclusions?

**Comment by denis-games** [comment]
@BiSerge I would like to do so, but I don't think I have the skills needed to conduct tests.

I though that I've publish sereval days my config, but I can't find the comment anymore.

- Nvidia GeForce GT 750M
- Intel Core i7-4500U

I use Godot 4.1.1 and with compatibility.

The only thing I've noticed so far, is that, using Xorg instead on Wayland on my 2013 Dell Inspiron 15 (dell inspiron 15 7000) has reduced the number of crashes.

**Comment by denis-games** [comment]
Sorry for the double post, but here is my configuration with trouble.

Computer model (purchased in 2013 and motherboard changed in 2019)
`Dell Inspiron 7537`

CPU
`Intel(R) Core(TM) i7-4500U CPU @ 1.80GHz`

GPU:
```
Haswell-ULT Integrated Graphics Controller
GK107M [GeForce GT 750M]
```

Godot with GPU
```
Godot Engine v4.1.1.stable.official.bd6af8e0e - https://godotengine.org
OpenGL API 4.6 (Core Profile) Mesa 23.0.4-0ubuntu1~23.04.1 - Compatibility - Using Device: Intel - Mesa Intel(R) HD Graphics 4400 (HSW GT2)
```

As said before, on both Xorg or Wayland, Godot is crashing, but less often on Xorg.

```
xserver-xorg-core: 2:21.1.7-1ubuntu3
apt-cache show mutter (Wayland): 44.3-0ubuntu1
```

**Comment by BiSerge** [comment]
I would like to get the help of developers in solving the problem.
